applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -39,830 +37,711 @@
                 free_size++;
-            }
 
         // Calculating the total size and required offset
         //int fixed_offset;
         int free_offset;
         int global_size;
 
-        // The correspounding offset by the sum of the sizes in thread with inferior rank
-        MPI_Scan(&free_size, &free_offset, 1, MPI_INT, MPI_SUM, MPI_COMM_WORLD);
+        // The correspounding offset by the sum of the sizes in thread with inferior current_rank
+        free_offset = r_data_comm.ScanSum(free_size);
 
         // The total size by the sum of all size in all threads
-        MPI_Allreduce(&free_size, &global_size, 1, MPI_INT, MPI_SUM, MPI_COMM_WORLD);
+        global_size = r_data_comm.SumAll(free_size);
 
         // finding the offset for the begining of the partition
         free_offset -= free_size;
 
         // Now setting the equation id with .
-        for (typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin(); dof_iterator != BaseType::mDofSet.end(); ++dof_iterator)
-            if(dof_iterator->GetSolutionStepValue(PARTITION_INDEX) == rank)
-            {
-                dof_iterator->SetEquationId(free_offset++);
-//  				std::cout << rank << " : set eq. id for dof " << dof_iterator->Id() << " to " << dof_iterator->EquationId() << std::endl;
-            }
-
+        for (auto &dof : BaseType::mDofSet)
+            if (dof.GetSolutionStepValue(PARTITION_INDEX) == current_rank)
+                dof.SetEquationId(free_offset++);
 
         BaseType::mEquationSystemSize = global_size;
         mLocalSystemSize = free_size;
-        if(BaseType::GetEchoLevel()>0){
-            std::cout << rank << " : BaseType::mEquationSystemSize = " << BaseType::mEquationSystemSize << std::endl;
-            std::cout << rank << " : mLocalSystemSize = " << mLocalSystemSize << std::endl;
-            std::cout << rank << " : free_offset = " << free_offset << std::endl;
-            //std::cout << rank << " : fixed_offset = " << fixed_offset << std::endl;
-        }
+        KRATOS_INFO_IF_ALL_RANKS("TrilinosBlockBuilderAndSolver", BaseType::GetEchoLevel() > 0) << std::endl
+                                                                                                << current_rank << " : BaseType::mEquationSystemSize = " << BaseType::mEquationSystemSize << std::endl
+                                                                                                << current_rank << " : mLocalSystemSize = " << mLocalSystemSize << std::endl
+                                                                                                << current_rank << " : free_offset = " << free_offset << std::endl;
 
         //by Riccardo ... it may be wrong!
-        mFirstMyId = free_offset-mLocalSystemSize;
-        mLastMyId = mFirstMyId+mLocalSystemSize;
-
-        r_model_part.GetCommunicator().SynchronizeDofs();
-
-    }
-
-    void UpdateGhostDofs(ModelPart& rThisModelPart)
```

Comment:
wre u sure thiss is never needed? even if you have ghost dofs with no nodes?

Reply:
I think so .. 
In the firstly its not used ... 
Secondly there is this 
https://github.com/KratosMultiphysics/Kratos/blob/d0c5615dc9d48c3ac113d4f1c2f25b2188e48326/applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h#L615

Where the dofs are synced ...

And then all the tests are passing so ...

Reply:
ok